### DISCUSSION TITLE
[Proposal] Add a property CameraOrientation and event OnOrientationChanged to detect the Camera's Orientation

### DISCUSSION THREAD

**Discussion by sourcetonuts** [CommunityToolkit/Maui#2308] [New Feature Discussions]
### Feature Name
Camera Orientation Detection
Noob and my first proposal and effort.

### Link to discussion
None

### Progress Tracker
Repo forked and work is on a branch https://github.com/sourcetonuts/Maui/tree/mct/feature-cameraview-orientation
- [x] Android Implementation
- [x] iOS Implementation
- [ ] MacCatalyst Implementation
- [ ] Windows Implementation
- [ ] Tizen Implementation
- [ ] Unit Tests
- [x] Samples
- [ ] Documentation

### Summary
New property CameraOrientation which is automatically detected and an event OnOrientationChanged when that property changes w/ the user rotating the phone (device fixed in Portrait mode still changes CameraOrientation property).

### Detailed Design
Implementation for IOS and Android on branch.

### Usage Syntax
Not applicable

### Drawbacks
Detection does take some processing time.

### Alternatives
No response

### Unresolved Questions
No response

**Comment by bijington** [comment]
Is it possible to determine the device orientation if the app is locked to portrait?

Also what about macOS and Windows? I guess there might be a windows tablet that can have orientation?

**Reply by sourcetonuts** [reply]
@bijington Shawn, yes it is. That's what the code in my forked branch does for iOS and Android.
https://github.com/sourcetonuts/Maui/tree/mct/feature-cameraview-orientation
I haven't looked at MacOS or WIndows, but I think it's less likely to need it or useful.

**Reply by bijington** [reply]
Thanks. Does this have the potential to fix the orientation issues that have been reported in the toolkit? If so I'd be happy to consider this a bug fix rather than a feature proposal and we just go ahead with it

**Reply by klindberg-magaya** [reply]
I'm not sure but I don't think so. IT currently doesn't rotate the image to the user's perspective/rotation. One could imagine doing that or leaving that to the caller.

**Reply by klindberg-magaya** [reply]
I'll add ... yes it certainly has the potential to be a fix certainly an improvement IMO.

**Reply by bijington** [reply]
Oh my bad I was thinking this was part of your proposal for some reason https://github.com/CommunityToolkit/Maui/pull/2312

**Reply by klindberg-magaya** [reply]
we are close, this code detects the orientation of the camera (w/ device rotation off too). It could but currently doesn't rotate the image deliver on MediaCatpure (sp). That could probably be added, but I don't yet have that solution. I would be OK if the CameraView code performed that operation. not sure everyone would want that TBD.

**Reply by klindberg-magaya** [reply]
Hey Shaun, I think this proposal has value. The code is working on the branch I forked in my repo. I have modified to Sample App to show the orientation changes in one of the floating labels.

https://github.com/sourcetonuts/Maui/tree/mct/feature-cameraview-orientation

Having not submitted before, 
- do I push a my branch here? Porobably I don't have rights.
- make a PR and submit? If so, how?

Any suggestions or help are apprciated and I'm happy to change things.

**Reply by klindberg-magaya** [reply]
This commit you mention seems to be IOS/Mac specific bug https://github.com/CommunityToolkit/Maui/pull/2312

This proposal offers a new feature. It works well and isn't a big change

For me the only question is whether to rotate the image on 
void OnMediaCaptured(Stream imageData);

I could see value in doing that for the user or leaving up to the user to decide what they want.

**Reply by sourcetonuts** [reply]
@bijington any thought on my proposal. Since it's my first I'm curious how the process works?

**Reply by bijington** [reply]
Sorry I've been preoccupied. I'll try and read over it and get back to you soon
